src/components/structures/MatrixChat.js

```diff
@@ -896,6 +897,12 @@ module.exports = React.createClass({
         });
     },
 
+    _setStateFromSessionStore() {
+        this.setState({
+            userHasGeneratedPassword: Boolean(this._sessionStore.getCachedPassword()),
```

Comment:
the `...this.state` is a hangover from when LoggedInView and MatrixChat were one - it was a way to make sure I wasn't missing any important bits of state. Please make any new bits of state explicit, and if you see any others which aren't being passed through explicitly, fix them. We should probably get rid of `...this.state`.